src/python/tests/test_configuration_manager.py

```diff
@@ -0,0 +1,710 @@
+                             'default': '127.0.0.1'}}},
+            'IPv6': {'category_description': 'IPv6 type',
+                     'category_value': {
+                         'info': {
+                             'description': "IPv6 type with default '2001:db8::'",
+                             'type': 'IPv6',
+                             'default': '2001:db8::'}}},
+            'X509': {'category_description': 'X509 Certification',
+                     'category_value': {
+                         'info': {
+                             'description': "X509 Certification",
+                             'type': 'X509 certificate',
+                             'default': 'x509_certificate.cer'}}},
+            'password': {'category_description': 'Password Type',
+                         'category_value': {
+                             'info': {
+                                 'description': "Password Type with default ''",
+                                 'type': 'password',
+                                 'default': ''}}}
+        }
+        for category_name in data:
+            await create_category(category_name=category_name,
+                                  category_description=data[category_name]['category_description'],
+                                  category_value=data[category_name]['category_value'])
+        results = await get_all_category_names()
+        for result in results:
+            assert data[result[0].replace(" ", "")]['category_description'] == result[1]
+        await self.teardown_method()
+
+    async def test_register_interest(self):
+        """
+        Test that when register_interest is called, _registered_interests gets updated
+        :assert:
+            1. when index of keys list is 0, corresponding name is 'boolean'
+            2. the value for _register_interests['boolean'] is {'tests.callback'}
+        """
+        await self.setup_method()
+        register_interest(category_name='boolean', callback='tests.callback')
+        assert list(_registered_interests.keys())[0] == 'boolean'
+        assert _registered_interests['boolean'] == {'tests.callback'}
+        await self.teardown_method()
+
+    async def test_register_interest_error(self):
+        """
+        Test that error gets returned when either category_name or callback is None
+        :Assert:
+            Assert error message when category_name is None
+            Assert error message when callback is None
+            Assert error messages when both are None
+        """
+        await self.setup_method()
+        with pytest.raises(ValueError) as error_exec:
+            register_interest(category_name=None, callback='foglamp.callback')
+        assert "ValueError: Failed to register interest. category_name cannot be None" in (
+            str(error_exec))
+
+        with pytest.raises(ValueError) as error_exec:
+            register_interest(category_name='integer', callback=None)
```

Comment:
where is third test?

Reply:
?

Reply:
ignore this, as either of value missing results in error. so OR type check is good enough